src/neural/network_check.cc

```diff
@@ -136,36 +229,102 @@ class CheckComputation : public NetworkComputation {
+        options.GetOrDefault<float>("atol", kDefaultAbsoluteTolerance);
+    params_.relative_tolerance =
+        options.GetOrDefault<float>("rtol", kDefaultRelativeTolerance);
+
+    const auto parents = options.ListSubdicts();
+    if (parents.size() > 0) {
+      backendName1 = parents[0];
+      backend1_dict = options.GetSubdict(backendName1);
+      backendName1 =
+          backend1_dict.GetOrDefault<std::string>("backend", backendName1);
+    }
+    if (parents.size() > 1) {
+      backendName2 = parents[1];
+      backend2_dict = options.GetSubdict(backendName2);
+      backendName2 =
+          backend2_dict.GetOrDefault<std::string>("backend", backendName2);
+    }
+    if (parents.size() > 2) {
+      std::cerr << "Warning, cannot check more than two backends" << std::endl;
+    }
 
-  static constexpr int CHECK_PROBABILITY = 5;
+    std::cerr << "Working backend set to " << backendName1 << "." << std::endl;
+    std::cerr << "Reference backend set to " << backendName2 << "."
+              << std::endl;
+
+    workNet_ =
+        NetworkFactory::Get()->Create(backendName1, weights, backend1_dict);
+    checkNet_ =
+        NetworkFactory::Get()->Create(backendName2, weights, backend2_dict);
+
+    checkFrequency_ =
+        options.GetOrDefault<float>("freq", kDefaultCheckFrequency);
+    switch (params_.mode) {
+      case kCheckOnly:
+        std::cerr << std::scientific << std::setprecision(1)
+                  << "Check mode: check only with relative tolerance "
+                  << params_.absolute_tolerance << ", absolute tolerance "
+                  << params_.relative_tolerance << "." << std::endl;
+        break;
+      case kErrorDisplay:
+        std::cerr << "Check mode: error display." << std::endl;
+        break;
+      case kHistogram:
+        std::cerr << "Check mode: histogram." << std::endl;
+        break;
+    }
+    std::cerr << "Check rate: " << std::fixed << std::setprecision(0)
+              << 100 * checkFrequency_ << "%." << std::endl;
+  }
 
   std::unique_ptr<NetworkComputation> NewComputation() override {
-    bool check = Random::Get().GetInt(0, CHECK_PROBABILITY) == 0;
+    double draw = Random::Get().GetDouble(1.0);
+    bool check = draw < checkFrequency_;
     if (check) {
-      std::unique_ptr<NetworkComputation> refComp = refNet_->NewComputation();
+      std::unique_ptr<NetworkComputation> refComp = workNet_->NewComputation();
       std::unique_ptr<NetworkComputation> checkComp =
           checkNet_->NewComputation();
-      return std::make_unique<CheckComputation>(std::move(refComp),
+      return std::make_unique<CheckComputation>(params_, std::move(refComp),
                                                 std::move(checkComp));
     }
-    return refNet_->NewComputation();
+    return workNet_->NewComputation();
   }
 
  private:
-  std::unique_ptr<Network> refNet_;
+  CheckParams params_;
+  double checkFrequency_;
+  std::unique_ptr<Network> workNet_;
```

Comment:
work_net_ and check_net_, and check_frequency_;
Also would be nice to have a comment what frequency is measured in? Herz? :)